src/Servers/Kestrel/Core/src/Internal/ConfigurationReader.cs

```diff
@@ -184,7 +184,17 @@ private IEnumerable<EndpointConfig> ReadEndpoints()
 
     private static SslProtocols? ParseSslProcotols(IConfigurationSection sslProtocols)
     {
-        var stringProtocols = sslProtocols.Get<string[]>();
+        // Avoid trimming warning from IConfigurationSection.Get<string[]>()
+        string[]? stringProtocols = null;
+        var childrenSections = sslProtocols.GetChildren().ToArray();
```

Comment:
Do you need ToArray()?

Reply:
GetChildren is an IEnumerable. It would run multiple times to get length, then loop over results.

Reply:
This is the code it's basically replicating - https://source.dot.net/#Microsoft.Extensions.Configuration.Binder/ConfigurationBinder.cs,548

Reply:
Got it